src/uct/cuda/base/cuda_md.c

```diff
@@ -55,6 +56,59 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_base_detect_memory_type,
     return UCS_ERR_INVALID_ADDR;
 }
 
+static ucs_status_t uct_cuda_base_get_sys_dev(ucs_sys_device_t *sys_dev_p)
+{
+    CUdevice dev;
+    int attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuCtxGetDevice(&dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_DOMAIN_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.domain = (uint16_t)attrib;
+
+    if (UCS_OK != UCT_CUDADRV_FUNC(cuDeviceGetAttribute(&attrib,
+                    CU_DEVICE_ATTRIBUTE_PCI_BUS_ID, dev))) {
+        return UCS_ERR_IO_ERROR;
+    }
+    sys_dev_p->bus_id.bus = (uint8_t)attrib;
+
+    return UCS_OK;
+}
+
+ucs_status_t uct_cuda_base_mem_query(uct_md_h md, const void *addr,
+                                     size_t length,
+                                     uct_md_mem_attr_t *mem_attr_p)
+{
+    ucs_status_t status;
+
+    mem_attr_p->field_mask = 0;
```

Comment:
this is not how field mask works for query function: called sets it, and the implementation should fill what is set

Reply:
@yosefe This is inconsistent with how we define the field_ask in UCT API. https://github.com/openucx/ucx/blob/master/src/uct/api/uct.h#L1247
```

typedef struct uct_md_mem_attr {
    /**
     * Mask of valid fields in this structure, using bits from
     * @ref uct_md_mem_attr_t. Note that the field mask is
     * populated upon return from uct_md_mem_query and not set by user.
     * Subsequent use of members of the structure are valid after ensuring that
     * relevant bits in the field_mask are set.
     */
    uint64_t          field_mask;
```
I've seen that the usual field_mask use is that the caller sets interested bits and query returns filled members. We would have to change the API semantics if we decide to have caller have control of which members need to be populated. Should we do that?

Reply:
i missed this in API review. We need to fix the API definition. The way it's defined now, does not really provide backward compatibility.

Reply:
I'll fix both instances then.